openrouteservice/src/main/java/org/heigit/ors/util/CSVUtility.java

```diff
@@ -66,6 +66,37 @@ private CSVUtility() {}
         return lines;
     }
 
+    /**
+     * Read data from CSV file using the provided filename, seperator, quote character and whether there is a header
+     *
+     * @param file          CSV file to read from
+     * @param ignoreHeader        Whether to ignore the first row of the CSV
+     * @param customDelimiter        Provide a custom delimiter that seperates columns
+     * @param customQuoteChar        Provide a custom character surrounding quotes
+     * @return              An ArrayList (rows) of ArrayLists (columns values)
+     */
+    public static ArrayList<ArrayList<String>> readFile(String file, boolean ignoreHeader, char customDelimiter, char customQuoteChar) {
+        // Open the CSV file
+        String ln = "";
+
+        ArrayList<ArrayList<String>> lines = new ArrayList<>();
+        boolean headerRead = false;
+        try (BufferedReader br = new BufferedReader(new FileReader(file))){
+            while ((ln = br.readLine()) != null) {
+                if(ignoreHeader && lines.isEmpty() && !headerRead) {
+                    headerRead=true;
+                    continue;
+                }
+
+                lines.add(parseLine(ln,customDelimiter,customQuoteChar));
+            }
+        } catch (IOException ioe) {
+            LOGGER.error("Could not read from file: " + file);
```

Comment:
It seems to me that in case the file could not be read, the whole process should be aborted instead of returning an empty collection. Otherwise the following code might continue to run on nothing and produce results that, while they might not be wrong, might make the user think that file readings were fine. Is that a valid assumption?

Reply:
Removed as anyway not used.